RNTester/js/CameraRollView.js

```diff
@@ -25,105 +23,123 @@ var {
    */
-  groupTypes: PropTypes.oneOf([
-    'Album',
-    'All',
-    'Event',
-    'Faces',
-    'Library',
-    'PhotoStream',
-    'SavedPhotos',
-  ]),
+  groupTypes:
+    | 'Album'
+    | 'All'
+    | 'Event'
+    | 'Faces'
+    | 'Library'
+    | 'PhotoStream'
+    | 'SavedPhotos',
 
   /**
    * Number of images that will be fetched in one page.
    */
-  batchSize: PropTypes.number,
+  batchSize: number,
 
   /**
    * A function that takes a single image as a parameter and renders it.
    */
-  renderImage: PropTypes.func,
+  renderImage: GetPhotosEdge => React.Node,
 
   /**
    * imagesPerRow: Number of images to be shown in each row.
    */
-  imagesPerRow: PropTypes.number,
+
+  imagesPerRow: number,
 
   /**
    * The asset type, one of 'Photos', 'Videos' or 'All'
    */
-  assetType: PropTypes.oneOf(['Photos', 'Videos', 'All']),
-};
-
-var CameraRollView = createReactClass({
-  displayName: 'CameraRollView',
-  propTypes: propTypes,
-
-  getDefaultProps: function(): Object {
+  assetType: 'Photos' | 'Videos' | 'All',
+|}>;
+
+type State = {|
+  assets: Array<GetPhotosEdge>,
+  lastCursor: ?string,
+  noMore: boolean,
+  loadingMore: boolean,
+  dataSource: ListViewDataSource,
+|};
+
+class CameraRollView extends React.Component<Props, State> {
+  static defaultProps = {
+    groupTypes: 'SavedPhotos',
+    batchSize: 5,
+    imagesPerRow: 1,
+    assetType: 'Photos',
+    renderImage: function(asset: GetPhotosEdge) {
+      const imageSize = 150;
+      const imageStyle = [styles.image, {width: imageSize, height: imageSize}];
+      return <Image source={asset.node.image} style={imageStyle} />;
+    },
+  };
+
+  state = this.getInitialState();
+
+  getInitialState() {
     return {
-      groupTypes: 'SavedPhotos',
-      batchSize: 5,
-      imagesPerRow: 1,
-      assetType: 'Photos',
-      renderImage: function(asset) {
-        var imageSize = 150;
-        var imageStyle = [styles.image, {width: imageSize, height: imageSize}];
-        return <Image source={asset.node.image} style={imageStyle} />;
-      },
-    };
-  },
-
-  getInitialState: function() {
-    var ds = new ListView.DataSource({rowHasChanged: this._rowHasChanged});
-
-    return {
-      assets: ([]: Array<Image>),
-      groupTypes: this.props.groupTypes,
-      lastCursor: (null: ?string),
-      assetType: this.props.assetType,
+      assets: [],
+      lastCursor: null,
       noMore: false,
       loadingMore: false,
-      dataSource: ds,
+      dataSource: new ListView.DataSource({rowHasChanged: rowHasChanged}),
     };
-  },
+  }
 
   /**
    * This should be called when the image renderer is changed to tell the
    * component to re-render its assets.
    */
-  rendererChanged: function() {
-    var ds = new ListView.DataSource({rowHasChanged: this._rowHasChanged});
+  rendererChanged() {
+    const ds = new ListView.DataSource({rowHasChanged: rowHasChanged});
     this.state.dataSource = ds.cloneWithRows(
       // $FlowFixMe(>=0.41.0)
```

Comment:
I think the general rule of thumb should be: If you see a `$FlowFixMe`, try to see if you can remove it.

Pretty sure this is suppressing some valid flow errors. For example, `rowHasChanged` takes an array of `Image` components, but `this.state.assets` is of type `Array<GetPhotosEdge>`.